### PR TITLE
BAU: Add Github Actions to deploy stack

### PR DESCRIPTION
**Pull request by surajkumar**
Create GHA to deploy to AWS

### CODE REVIEW

File: .github/workflows/deploy-branch.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    outputs:
+      aws-region: ${{ steps.deploy.outputs.aws-region }}
+      stack-name: ${{ steps.deploy.outputs.stack-name }}
+    steps:
+      - name: Deploy stack
+        uses: govuk-one-login/github-actions/sam/deploy-stack@e6b6ed890b35904e1be79f7f35ffec983fa4d9db
+        id: deploy
+        with:
+          sam-deployment-bucket: ${{ vars.DEPLOYMENT_ARTIFACTS_BUCKET }}
+          aws-role-arn: ${{ vars.DEPLOYMENT_ROLE_ARN }}
```
Comment (mrdoodles) [inline .github/workflows/deploy-branch.yml:55]: This should be a secret and not a var

File: .github/workflows/deploy-branch.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      url: ${{ steps.deploy.outputs.stack-url }}
+    outputs:
+      aws-region: ${{ steps.deploy.outputs.aws-region }}
+      stack-name: ${{ steps.deploy.outputs.stack-name }}
+    steps:
+      - name: Deploy stack
+        uses: govuk-one-login/github-actions/sam/deploy-stack@e6b6ed890b35904e1be79f7f35ffec983fa4d9db
+        id: deploy
+        with:
+          sam-deployment-bucket: ${{ vars.DEPLOYMENT_ARTIFACTS_BUCKET }}
```
Comment (mrdoodles) [inline .github/workflows/deploy-branch.yml:54]: This should be a secret

File: .github/workflows/clean-up-deployments.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+jobs:
+  delete-stacks:
+    name: Delete stale stacks
+    runs-on: ubuntu-latest
+    environment: development
+    steps:
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.DEPLOYMENT_ROLE_ARN }}
```
Comment (mrdoodles) [inline .github/workflows/clean-up-deployments.yml:25]: This should be a secret

File: .github/workflows/delete-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          usage: Stack name
+          prefix: preview-check-hmrc-api
+          length-limit: 128
+          verbose: false
+
+      - name: Delete stack
+        uses: govuk-one-login/github-actions/sam/delete-stacks@e6b6ed890b35904e1be79f7f35ffec983fa4d9db
+        with:
+          stack-names: ${{ steps.get-stack-name.outputs.pretty-branch-name }}
+          aws-role-arn: ${{ vars.DEPLOYMENT_ROLE_ARN }}
```
Comment (mrdoodles) [inline .github/workflows/delete-deployment.yml:35]: This should be a secret

File: .github/workflows/post-merge-deploy-to-build.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      id-token: write
+      contents: read
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.BUILD_GH_ACTIONS_ROLE_ARN }}
```
Comment (mrdoodles) [inline .github/workflows/post-merge-deploy-to-build.yml:25]: This should be a secret

File: .github/workflows/delete-deployment.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+          usage: Stack name
+          prefix: preview-check-hmrc-api
+          length-limit: 128
+          verbose: false
+
+      - name: Delete stack
+        uses: govuk-one-login/github-actions/sam/delete-stacks@e6b6ed890b35904e1be79f7f35ffec983fa4d9db
+        with:
+          stack-names: ${{ steps.get-stack-name.outputs.pretty-branch-name }}
+          aws-role-arn: ${{ secrets.DEPLOYMENT_ROLE_ARN }}
```
Comment (CharlesIC) [inline .github/workflows/delete-deployment.yml]: ARNs are not secret. These should be reverted back to variables like in other repos.

File: .github/workflows/delete-deployment.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    environment: development
+
+    steps:
+      - name: Get stack name
+        uses: govuk-one-login/github-actions/beautify-branch-name@e6b6ed890b35904e1be79f7f35ffec983fa4d9db
+        id: get-stack-name
+        with:
+          usage: Stack name
+          prefix: st
+          length-limit: 128
```
Comment (CharlesIC) [inline .github/workflows/delete-deployment.yml]:
```suggestion
          length-limit: 14
```
This needs to match `deploy-branch.yml`

File: .github/jest.config.ci.ts
Comment (CharlesIC) [inline .github/jest.config.ci.ts:1]: This file is unused since we don't run any tests in GHA, so it can be deleted

**Review by CharlesIC — APPROVED** [review]

**Pull request merged**